src/block/block_ext.c

```diff
@@ -390,6 +397,7 @@ __block_off_remove(
 
     /* Update the cached end-of-list. */
     if (el->last == ext)
+        /* Request a lazy update of el->last in __block_append. */
```

Comment:
This comment isn't accurate.

This code isn't really "requesting" an update of the cached end-of-list. It's just clearing it because we're removing that element and we don't know what the preceding element is. (i.e., it's not worth the cost of keeping track of it in case we need it here). 

There are several functions that can update it, not just `__block_append()`. 

If you think the comment needs to be clearer here maybe at line #398 say "Clear the cached end-of-list if we are removing the last element."

Reply:
The reason I put this comment in here is the external PR for __block_off_remove not updating WT_EXTLIST.last correctly. It is actually a case of lazy update but the person did not realize that. If this comment had been here, perhaps that incorrect PR never would have happened.

I checked.
* Only one function reads WT_EXTLIST.last: __block_append.
* Only one function checks WT_EXTLIST.last == NULL and initializes it if so: __block_append.
* Only one function clears WT_EXTLIST.last: __block_off_remove. All other functions that set WT_EXTLIST.last set it to the correct value.

It seems to me that whenever something is done lazily something has to mark that it needs to be done, and something else needs to do it if not done already.

I tried to convey this in my comments showing where it is marked needed, and where it is done if not done already.

I changed this comment to: "To save time, update to correct value later."
I deleted the comment in __block_append. A vague comment such as "Not set, update now." is too obvious.